code/centauri/hyperspace/parachain/src/chain.rs

```diff
@@ -176,4 +395,50 @@ where
 
 		Ok(())
 	}
+
+	async fn query_client_message(
```

Comment:
@seunlanlege  could we just store the client update headers offchain, so they can be retrieved is an easier manner.

Reply:
Already discussed with @vmarkushin , we'll use tx querying as that is the most chain agnostic solution.